editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx

```diff
@@ -401,6 +401,36 @@ export function renderCoreElement(
         elementPath,
       )
 
+      const elementIsTextEdited = elementPath != null && EP.pathsEqual(elementPath, editedText)
+
+      if (elementIsTextEdited) {
+        const text = trimAndJoinTextFromJSXElements([actualElement])
+        const textContent = unescapeHTML(text ?? '')
+        const textEditorProps: TextEditorProps = {
+          elementPath: elementPath,
+          filePath: filePath,
+          text: textContent,
+          component: React.Fragment,
+          passthroughProps: {},
+          textProp: activeConditionValue ? 'whenTrue' : 'whenFalse',
+        }
+
+        return buildSpyWrappedElement(
+          actualElement,
+          textEditorProps,
+          childPath!,
+          metadataContext,
+          updateInvalidatedPaths,
+          [],
+          TextEditorWrapper,
+          inScope,
+          jsxFactoryFunctionName,
+          shouldIncludeCanvasRootInTheSpy,
+          imports,
+          filePath,
+        )
```

Comment:
is this a copy-paste of the other `if (elementIsTextEdited)` block? if so, could we turn this into a shared function? I worry that we'll fix an unescape bug in one path but not the other